dev/Repeater/APITests/ViewManagerTests.cs

```diff
@@ -586,6 +586,43 @@ public void CanResetLayoutAfterUniqueIdReset()
             });
         }
 
+        [TestMethod]
+        public void ValidateDataContextDoesNotGetOverwritten()
+        {
+            ItemsRepeater repeater = null;
+            const string c_element1DataContext = "Element1_DataContext";
+
+            RunOnUIThread.Execute(() =>
+            {
+                List<Button> data = new List<Button>();
+
+                var element1 = new Button();
+                element1.Content = "Element1_Content";
+                element1.DataContext = c_element1DataContext;
+                data.Add(element1);
+
+                var elementFactory = new DataAsElementElementFactory();
+
+                repeater = new ItemsRepeater() {
+                    ItemsSource = data,
+                    ItemTemplate = elementFactory,
+                    Layout = new StackLayout()
+                };
+
+                Content = repeater;
+            });
+
+            IdleSynchronizer.Wait();
+
+            RunOnUIThread.Execute(() =>
+            {
```

Comment:
You can remove these two separate UIExecutor blocks and just do a Content.UpdateLayout() instead. If you do that you can also bring in the ItemsRepeater declaration to where it is created.